framework/web_app/src/pages/StockOverview.vue

```diff
@@ -1,96 +1,260 @@
-</script>
-
 <template>
-    <form @submit.prevent="test">
-        <label for="name">Name</label>
-        <input id="name" v-model="form.name" />
-
-        <br />
-
-        <label for="stockLevel">Stock Level</label>
-        <select id="stockLevel" v-model="form.stock_level_id">
-            <option v-for="stockLevel in stockLevels" :key="stockLevel.stock_level_id" :value="stockLevel.stock_level_id">
-                {{ stockLevel.description }}
-            </option>
-        </select>
-
-        <br />
-
-        <!-- <input type="email" v-model="form.a" />
-
-        <label>Message is: {{ message }}</label>
-        <input v-model="message" placeholder="edit me" />
-
-
-        <textarea v-model="form.b" />
-
-
-        <select v-model="form.c">
-            <option value="new-york">New York</option>
-            <option value="moscow">Moscow</option>
-        </select>
-
-        <input type="checkbox" v-model="form.d" />
-
-        <input type="radio" value="weekly" v-model="form.e" />
-        <input type="radio" value="monthly" v-model="form.f" /> -->
-
-        <button type="submit">Submit</button>
-    </form>
-    <!-- <button type="button" class="btn btn-success" @click="testApi">{{ test }}</button> -->
-    <table class="table table-hover">
-        <thead>
-            <tr>
-                <th scope="col">Name</th>
-                <th scope="col">Action</th>
-            </tr>
-        </thead>
-        <tbody>
-            <tr v-for="stockItem in stockItems" :key="stockItem.stock_item_id">
-                <td>{{ stockItem.name }}</td>
-                <td>
-                    <div class="btn-group" role="group">
-                        <button type="button" class="btn btn-warning btn-sm">Edit</button>
-                        <button type="button" class="btn btn-danger btn-sm">Delete</button>
-                    </div>
-                </td>
-            </tr>
-        </tbody>
-    </table>
+    <q-btn
+        class="q-ma-sm"
+        @click="shouldDisplayCreateStockItemModal = true"
+        color="green"
+    >
+        <q-icon
+            name="add"
+            color="dark-green"
+            size="30px"
+        />
+    </q-btn>
+
+    <q-card
+        class="no-shadow q-ma-sm"
+        :key="item.stock_item_id"
+        bordered
+        v-for="item in stockItems"
+        vertical="false"
+    >
+        <q-card-section
+            class="row justify-between"
+            horizontal
+        >
+            <q-card-actions class="col">
+                <q-btn-dropdown
+                    class="q-mx-sm"
+                    :color="getStockLevelColour(item.stock_level_id)"
+                    :items="stockLevelStore.stockLevels"
+                    dense
+                    dropdown-icon="none"
+                    no-caps
+                    push
+                    rounded
+                    style="width: 28px"
```

Comment:
At some point i'll go over all of the styling and move the common ones into app.scc to be shared classes.

Reply:
After this PR